b2sdk/v2/version_utils.py

```diff
@@ -0,0 +1,46 @@
+######################################################################
+#
+# File: b2sdk/v2/version_utils.py
+#
+# Copyright 2023 Backblaze Inc. All Rights Reserved.
+#
+# License https://www.backblaze.com/using_b2_code.html
+#
+######################################################################
+
+from b2sdk import _v3 as v3
+
+
+class _OldAbstractVersionDecoratorMixin:
+    def __call__(self, *args, **kwargs):
+        if self.cutoff_version:
+            assert self.current_version < self.cutoff_version, '%s decorator is still used in version %s when old %s name %r was scheduled to be dropped in %s. It is time to remove the mapping.' % (
+                self.__class__.__name__,
+                self.current_version,
+                self.WHAT,
+                self.source,
+                self.cutoff_version,
+            )
+        return super().__call__(*args, **kwargs)
+
+
+class _OldAbstractDeprecatorMixin(_OldAbstractVersionDecoratorMixin):
+    def __call__(self, *args, **kwargs):
+        super(_OldAbstractVersionDecoratorMixin, self).__call__(*args, **kwargs)
```

Comment:
Why not just putting code of `_OldAbstractVersionDecoratorMixin` here? Or if you want logic separation, you could create a method, not a mixin.